assets/src/components/Tasks.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from "react";
+import fetchTasks from './tasks/request'
+import Task from './tasks/Task'
+
+type State = [] | [{technology: string, counter: number}]
+
+function TasksIndex() {
+  const [tasks, setTasks] = useState<State>([])
+
+  useEffect(() => {
+    function doFetchTasks() {
+      fetchTasks().then(response => {
+        if ('tasks' in response) {
+          setTasks(response.tasks)
+        } else {
+          // Handle error
```

Comment:
The code LGTM, I just need to check one thing here.

We should return Result as Uptight Result and just print the assertion error + trace information in errors by default.

Make sure we're doing that.

Then, when we're going to be making the errors prettier, we'll have a dictionary function that takes an error and "translates" it depending on who is looking at the error. If it's the user, it's gonna be a "translation" from English to English, perhaps, if it's a developer, it's gonna be an identity function.

Reply:
Sure, it *is* working like that for commands submission, but here the only error I expect to see is `Unauthorized` (when session will expire) and we want to just `window.location('/login')` here for example. In this case there will no be any Uptight Result returned.